### PR TITLE
Redesign/line counter

### PR DESCRIPTION
**Pull request by hardikdava**
# Description

Colab link: https://colab.research.google.com/drive/1k___Gmq_hT79VQsh2prHUdL4D38IXtbm?usp=sharing

Features/Improvement:

 - [x] Multi-Class counting #93  #94 
 - [x] Extra counts bug #87 
 - [x] Text color for `in/out` #79 
 - [x] API to get info of `in/out` counts for more modification for users along track ids
 - [x] objects anchor selection for counting
 - [x] objects anchor point drawing
 - [x] change line color on crossing
 - [x] return counts api and total counts api
 - [x] return functions to get crossed detections for `in` and `out`
 - [x] introduce `sv.AnchorPointAnnotator` to render points on image
 - [ ] Improve Documentation

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

**Comment by hardikdava** [comment]
@SkalskiP Do you think of any other features to be included? Please go through the descriptions once.

File: supervision/application/line_counter.py [hunk 1]
```diff
@@ -1,74 +1,135 @@
...
-    def __init__(self, start: Point, end: Point):
+    def __init__(self, start: Point, end: Point, anchor: Position = Position.CENTER):
         """
         Initialize a LineCounter object.
 
         Attributes:
             start (Point): The starting point of the line.
             end (Point): The ending point of the line.
-
+            anchor (Position): The position of the anchor point of detectios for counting
```
Comment (AlainPilon) [inline supervision/application/line_counter.py:23]: typo: detectios => detections

Comment (hardikdava) [reply supervision/application/line_counter.py]: Thanks for finding out.

**Comment by hardikdava** [comment]
hello @SkalskiP @onuralpszr , let's try to finish this as it will make good reflection from community.

**Comment by onuralpszr** [comment]
@hardikdava  I am starting to review

**Comment by onuralpszr** [comment]
@hardikdava  I accept your changes for AnchorAnnotator and PolygonZone Annotator which we need to change both later on.

**Comment by onuralpszr** [comment]
@hardikdava  we merge new Annotators now  I am going to rebase this and do proper changes and do another round of review, I am started working on it.

**Comment by onuralpszr** [comment]
> @onuralpszr @SkalskiP Should we include this in next release?

I was working on this and going send commits and hopefully this one and other line (in/out) improvement one I want to for next release as well.

**Comment by SkalskiP** [comment]
@hardikdava I'm not sure I will have enough time to look at it this week :/

**Comment by SkalskiP** [comment]
Hi @hardikdava 👋🏻, I need to ask you to refactor the `trigger` method and make it more readable. Please move different parts of the logic to separate functions with descriptive names. It would also be awesome if you could describe what is happening there in a comment.

**Comment by AlainPilon** [comment]
He. Is it going to be part of the next release? Is is abandoned and will need a full review/rewrite? I have delayed using the line counter for as long as I could but I will need to implement this in our project soon so I would love to have some visibility on this PR.

**Pull request closed without merging**